dashboard/app/models/regional_partner.rb

```diff
@@ -79,6 +79,17 @@ def future_pd_workshops_organized
     pd_workshops_organized.future
   end
 
+  def earliest_summer_workshop_apps_open_date
+    Date.parse([apps_open_date_csd_teacher, apps_open_date_csp_teacher].max).strftime('%B %e, %Y')
+  end
+
+  def upcoming_summer_workshops
+    pd_workshops.
+      future.
+      select {|w| w.local_summer? && w.effective_num_days >= 5}.
```

Comment:
This `select` with block will retriever and iterate through the full set of results from the query, all future workshops associated with this partner. Granted that's not likely to be too huge, but still I think we can offload this to Sql. 

Do we need the `effective_num_days` check? I'm fairly certain that all summer workshops are 5 days. I'd suggest querying [local_summer?](https://github.com/code-dot-org/code-dot-org/blob/36ec49ce078e00df5da8cb48dcbe88d01c087516/dashboard/app/models/pd/workshop.rb#L515) with a where clause and either dropping the `effective_num_days` check entirely, or moving that into its own select block after the filter:

```ruby
  where(subject: Pd::Workshop::SUBJECT_SUMMER_WORKSHOP).
  select {|w| w.effective_num_days >= 5}. # this line is probably not needed
  map...
```